### PR TITLE
Py36 update, Black and pyupgrade in pre-commit

### PR DESCRIPTION
**Pull request by mshriver**
Consistency with tooling and f-string use in airgun and robottelo.

Using pyupgrade and black in pre-commit to auto-format and apply some f-string conversions.

I manually converted all the uses of `.format()` and string 'addition' that I found.

### CODE REVIEW

**Review by pondrejk — COMMENTED** [review]
uf, did my best to gaze through it, ack pending travis

File: nailgun/entities.py [hunk 1]
```diff
@@ -747,21 +730,19 @@ def path(self, which=None):
...
 
         """
-        if which and which.startswith('content_'):
-            return '{0}/content/{1}'.format(
-                super(Capsule, self).path(which='self'),
-                which.split('content_')[1]
-            )
-        return super(Capsule, self).path(which)
+        if which and which.startswith("content_"):
+            return '{super().path(which="self")}/content/{which.split("content_")[1]}'
```
Comment (ogajduse) [inline nailgun/entities.py:734]:
```suggestion
            return f'{super().path(which="self")}/content/{which.split("content_")[1]}'
```

**Review by ogajduse — CHANGES_REQUESTED** [review]
Wow, that's huge! :hot_face:
One change suggested. This should resolve CI failures.

**Review by ogajduse — APPROVED** [review]

**Review by ogajduse — CHANGES_REQUESTED** [review]
I see one DeprecationWarning in the travis [build](https://travis-ci.org/github/SatelliteQE/nailgun/jobs/740896898#L393). Can you please incorporate the fix into your PR?

https://github.com/mshriver/nailgun/blob/28d312a699f453c0b3b75f8d3f29bde5c4742894/tests/test_entities.py#L2071

**Review by ogajduse — DISMISSED** [review]

**Comment by jyejare** [comment]
@SatelliteQE/airgun-tier-2-reviewers Reviewing this PR as Tier2 reviewer.

File: nailgun/entities.py [hunk 2]
```diff
@@ -7298,22 +6911,26 @@ def __init__(self, server_config=None, **kwargs):
...
     """A representation of a Subnet entity."""
 
     def __init__(self, server_config=None, **kwargs):
         self._fields = {
             'boot_mode': entity_fields.StringField(
-                choices=('Static', 'DHCP',),
+                choices=(
+                    'Static',
+                    'DHCP',
+                ),
```
Comment (jyejare) [inline nailgun/entities.py:6933]: The earlier one was much better with some change as:
```suggestion
                choices=( 'Static',  'DHCP'),
```

File: nailgun/entity_mixins.py [hunk 3]
```diff
@@ -177,14 +177,11 @@ def _payload(fields, values):
     for field_name, field in fields.items():
         if field_name in values:
             if isinstance(field, OneToOneField):
-                values[field_name + '_id'] = (
-                    getattr(values.pop(field_name), 'id', None)
-                )
+                values[field_name + '_id'] = getattr(values.pop(field_name), 'id', None)
```
Comment (jyejare) [inline nailgun/entity_mixins.py:180]:
```suggestion
                values[f'{field_name}_id'] = getattr(values.pop(field_name), 'id', None)
```

File: nailgun/entity_mixins.py [hunk 4]
```diff
@@ -177,14 +177,11 @@ def _payload(fields, values):
...
             elif isinstance(field, OneToManyField):
-                values[field_name + '_ids'] = [
-                    entity.id for entity in values.pop(field_name)
-                ]
+                values[field_name + '_ids'] = [entity.id for entity in values.pop(field_name)]
```
Comment (jyejare) [inline nailgun/entity_mixins.py:182]: same as above

**Review by jyejare — COMMENTED** [review]
Its a B-I-G Pull request. :)

Since all the unit tests are passing, I wont ask to run Tier1 Tests for this huge refactor. Kudos to the unit tests and to mshriver. unittests ++
@mshriver ++

ACK with some minor comments.

File: docs/create_user_nailgun.py [hunk 5]
```diff
@@ -13,9 +11,9 @@ def main():
     """Create an identical user account on a pair of satellites."""
     server_configs = ServerConfig.get('sat1'), ServerConfig.get('sat2')
     for server_config in server_configs:
-        org = Organization(server_config).search(
-            query={'search': 'name="Default_Organization"'}
-        )[0]
+        org = Organization(server_config).search(query={'search': 'name="Default_Organization"'})[
+            0
+        ]
```
[the remark this answers is not in the document]
Comment (JacobCallahan) [reply docs/create_user_nailgun.py:16]: this is likely black's formatting

File: nailgun/entity_mixins.py (hunk 3 above)
Comment (mshriver) [reply nailgun/entity_mixins.py:180]: Apply this in several places

File: nailgun/entities.py (hunk 2 above)
Comment (mshriver) [reply nailgun/entities.py:6933]: This is all formatted by Black, can't pick which of these we like an don't like.

File: nailgun/entities.py [hunk 6]
```diff
@@ -1283,33 +1234,31 @@ def update(self, fields=None):
...
-    # Refresh external usergroup
-    ExternalUserGroup(id=<id>, usergroup=usergroup).refresh()
+     # Create external usergroup
+     ExternalUserGroup(name='foobargroup',usergroup=usergroup,auth_source=auth).create()
+     # Read external usergroup
+     ExternalUserGroup(id=<id>, usergroup=usergroup).read()
+     # Delete external usergroup
+     ExternalUserGroup(id=<id>, usergroup=usergroup).delete()
+     # Refresh external usergroup
+     ExternalUserGroup(id=<id>, usergroup=usergroup).refresh()
```
[the remark this answers is not in the document]
Comment (mshriver) [reply nailgun/entities.py:1259]: The previous indentation was wrong, this was formatted by Black

**Review by JacobCallahan — CHANGES_REQUESTED** [review]

File: docs/create_user_plain.py [hunk 7]
```diff
@@ -59,18 +57,18 @@ def get_organization_id(server_config, label):
...
-        server_config['url'] + '/katello/api/v2/organizations',
-        data=json.dumps({'search': 'label={}'.format(label)}),
-        auth=server_config['auth'],
-        headers={'content-type': 'application/json'},
-        verify=server_config['verify'],
+        f'{server_config["url"]}/katello/api/v2/organizations',
+        data=json.dumps({"search": f"label={label}"}),
+        auth=server_config["auth"],
+        headers={"content-type": "application/json"},
+        verify=server_config["verify"],
```
[the remark this answers is not in the document]
Comment (mshriver) [reply docs/create_user_plain.py:64]: Kinda looks like it, not exactly sure why.

File: tests/test_entity_mixins.py [hunk 8]
```diff
@@ -449,13 +415,11 @@ def test_compare(self):
         alice_2 = SampleEntity(self.cfg, id=2, name='Alice', unique='b')
         self.assertTrue(
             alice.compare(alice_2),
-            'Both "id" and "unique" are unique fields, thus must be ignored '
-            'compare by default'
+            'Both "id" and "unique" are unique fields, thus must be ignored ' 'compare by default',
```
[the remark this answers is not in the document]
Comment (mshriver) [reply tests/test_entity_mixins.py:418]: Searching for and fixing all of these, good eye.

File: nailgun/entities.py [hunk 9]
```diff
@@ -4786,9 +4567,8 @@ def create_payload(self):
         <https://bugzilla.redhat.com/show_bug.cgi?id=1238757>`_.
 
         """
-        payload = super(LifecycleEnvironment, self).create_payload()
-        if (_get_version(self._server_config) < Version('6.1') and
-                'prior_id' in payload):
+        payload = super().create_payload()
+        if _get_version(self._server_config) < Version('6.1') and 'prior_id' in payload:
```
[the remark this answers is not in the document]
Comment (mshriver) [reply nailgun/entities.py:4571]: https://github.com/SatelliteQE/nailgun/pull/767    Like this?

**Review by JacobCallahan — APPROVED** [review]
ACK

**Comment by mshriver** [comment]
This will get one more rebase after #766

File: docs/create_user_nailgun.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (jyejare) [reply docs/create_user_nailgun.py:16]: @JacobCallahan Yes! But it looks weird and this is easy manual fix though !

**Review by jyejare — APPROVED** [review]

**Review by JacobCallahan — APPROVED** [review]

**Pull request merged**